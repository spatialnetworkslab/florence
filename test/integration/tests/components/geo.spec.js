/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/geo')
  })

  describe('Geo Tests', () => {
    it('[render] polygons rendered', () => {
      cy.get('.polygon').should('have.length', 3)
    })
    it('[render] geo polygons correctly', () => {
      cy.get('svg')
        .snapshot({ name: 'svg with 3 geo polygons' })
    })
  })
})
