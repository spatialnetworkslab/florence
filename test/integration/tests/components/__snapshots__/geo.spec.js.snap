exports[`Assertions > Geo Tests > [render] geo polygons correctly #0`] = `
<svg
  width="500"
  height="500"
  viewBox="0 0 500 500"
  preserveAspectRatio="xMidYMid meet"
>
  <defs>
    <mask id="gr0-mask-padding-bg">
      <rect x="0" y="0" width="500" height="500" fill="white"></rect>
      <rect x="0" y="0" width="500" height="500" fill="black"></rect>
    </mask>
  </defs>
  <defs>
    <clipPath id="clip-sc0">
      <rect x="0" y="0" width="500" height="500"></rect>
    </clipPath>
    <mask id="mask-sc0-padding-bg">
      <rect x="0" y="0" width="500" height="500" fill="white"></rect>
      <rect x="0" y="0" width="500" height="500" fill="black"></rect>
    </mask>
  </defs>
  <g class="section" clip-path="url(#clip-sc0)">
    <g class="polygon-layer">
      <path
        class="polygon"
        d="M0,333.33L500,333.33L500,166.67L0,166.67Z"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="0.5"
      ></path>
      <path
        class="polygon"
        d="M166.67,333.33L250,291.67L83.33,291.67Z"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="0.5"
      ></path>
      <path
        class="polygon"
        d="M333.33,166.67L250,208.33L416.67,208.33Z"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="0.5"
      ></path>
    </g>
  </g>
</svg>
`;
